modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -205,15 +232,19 @@ private Replica startReplicaInternal(
             ReplicationGroupId replicaGrpId,
             ReplicaListener listener
     ) {
-        var replica = new Replica(replicaGrpId, listener);
+        replicas.compute(replicaGrpId, (replicationGroupId, replicaFut) -> {
+            if (replicaFut == null) {
+                replicaFut = CompletableFuture.completedFuture(new Replica(replicaGrpId, listener));
 
-        Replica previous = replicas.putIfAbsent(replicaGrpId, replica);
+                return replicaFut;
+            } else {
+                replicaFut.complete(new Replica(replicaGrpId, listener));
 
-        if (previous != null) {
-            throw new ReplicaIsAlreadyStartedException(replicaGrpId);
-        }
+                return replicaFut;
+            }
+        });
 
-        return replica;
+        return replicas.get(replicaGrpId).join();
```

Comment:
Let's add a comment that replicaFut is always completed here.

Reply:
fixed